MekHQ/src/mekhq/campaign/finances/enums/FinancialTerm.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (C) 2021 - The MegaMek Team. All Rights Reserved.
+ *
+ * This file is part of MekHQ.
+ *
+ * MekHQ is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * MekHQ is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MekHQ. If not, see <http://www.gnu.org/licenses/>.
+ */
+package mekhq.campaign.finances.enums;
+
+import megamek.common.util.EncodeControl;
+import mekhq.MekHQ;
+
+import java.time.DayOfWeek;
+import java.time.LocalDate;
+import java.time.temporal.TemporalAdjusters;
+import java.util.ResourceBundle;
+
+public enum FinancialTerm {
+    //region Enum Declarations
+    BIWEEKLY("FinancialTerm.BIWEEKLY.text", "FinancialTerm.BIWEEKLY.toolTipText"),
+    MONTHLY("FinancialTerm.MONTHLY.text", "FinancialTerm.MONTHLY.toolTipText"),
+    QUARTERLY("FinancialTerm.QUARTERLY.text", "FinancialTerm.QUARTERLY.toolTipText"),
+    ANNUALLY("FinancialTerm.ANNUALLY.text", "FinancialTerm.ANNUALLY.toolTipText");
+    //endregion Enum Declarations
+
+    //region Variable Declarations
+    private final String name;
+    private final String toolTipText;
+    //endregion Variable Declarations
+
+    //region Constructors
+    FinancialTerm(final String name, final String toolTipText) {
+        final ResourceBundle resources = ResourceBundle.getBundle("mekhq.resources.Finances", new EncodeControl());
+        this.name = resources.getString(name);
+        this.toolTipText = resources.getString(toolTipText);
+    }
+    //endregion Constructors
+
+    //region Getters
+    public String getToolTipText() {
+        return toolTipText;
+    }
+    //endregion Getters
+
+    //region Boolean Comparison Methods
+    public boolean isBiweekly() {
+        return this == BIWEEKLY;
+    }
+
+    public boolean isMonthly() {
+        return this == MONTHLY;
+    }
+
+    public boolean isQuarterly() {
+        return this == QUARTERLY;
+    }
+
+    public boolean isAnnually() {
+        return this == ANNUALLY;
+    }
+    //endregion Boolean Comparison Methods
+
+    public LocalDate nextValidDate(final LocalDate origin) {
```

Comment:
Valid for what?

Reply:
Valid for the next financial term... so either a payment is due or a payment made to the company at that date